### PR TITLE
BibAuthorID: spin in connect with hepnames form

### PR DESCRIPTION
**Pull request by MSusik**
- Fixes issue with duplicated RT tickets.

Signed-off-by: Mateusz Susik mateusz.susik@cern.ch

### CODE REVIEW

**Comment by MSusik** [comment]
The problem was, that user was not notified about the request sent and he was able to click on `Yes, go ahead` button many times. I have hidden the button after the first click and added a spin there.

**Comment by MSusik** [comment]
@kaplun

**Comment by kaplun** [comment]
Great! Thanks. I am not enough expert on bootstrap to be able to understand your code. Maybe @jmartinm can help me in the review?

**Comment by MSusik** [comment]
Actually, this is more `js` than `bootstrap`. I will explain shortly.

If you go to author's profile and want to connect profile with Hepnames record, you will see a modal with `Yes, go ahead` button. In the content it says sth like `Do you really want to connect this author's profile with blablabla`. After user clicks it, the AJAX request is sent (`"show.bs.modal"` function). Before the changes, all the buttons were visible for the user during the process of sending the ticket to RT. I don't know why, but `bibcatalog` function takes 3-4 seconds. A lot of users were confused and were clicking `Yes, go ahead` many times. After the ticket is sent, the response is sent and `successCallback` or `errorCallback` is run. Those functions change the content of the box.

I don't know why Minn implemented it this way instead of creating two different boxes. Anyway, the simpliest fix was to add spin inside `"show.bs.modal"`. Those lines do it.

``` js
$modal.find(".modal-body").html('<div style="height:200px">' +
                    '<span id="hepSpin" style="position: absolute;display: block;top: 50%;left: 50%;">' +
                    '</span></div>');
var spin = new Spinner({length: 40, radius:40, width:20, corners: 0.8}).spin().el;
$modal.find("#hepSpin").append(spin);
```

File: modules/bibauthorid/lib/bibauthorid.js [hunk 1]
```diff
@@ -98,20 +98,32 @@ var ticketbox = {
...
+                    anyCallback();
                     $modal.find(".modal-body").html("<p>There is a problem with INSPIRE at the moment. Please try again in a couple of minutes.</p>");
                     $modal.find(".modal-header>.modal-title").text("Try again later");
-                    $modal.find(".modal-footer>a.confirm").remove();
-                    $modal.find(".modal-footer>a.back").text("Close");
                   };
 
+                  $modal.find(".modal-body").html('<div style="height:200px">' +
+                    '<span id="hepSpin" style="position: absolute;display: block;top: 50%;left: 50%;">' +
+                    '</span></div>');
```
Comment (jmartinm) [inline modules/bibauthorid/lib/bibauthorid.js]: All the styling should go in `bibauthorid.css`, not inline.

File: modules/bibauthorid/lib/bibauthorid.js [hunk 2]
```diff
@@ -98,20 +98,32 @@ var ticketbox = {
...
+                  var spin = new Spinner({length: 40, radius:40, width:20, corners: 0.8}).spin().el;
+                  $modal.find("#hepSpin").append(spin);
+                  $modal.find(".modal-footer>a.confirm").remove();
+                  $modal.find(".modal-footer>a.back").hide();
+                  $modal.find(".modal-footer>a.back").text("Close");
```
Comment (jmartinm) [inline modules/bibauthorid/lib/bibauthorid.js]: Isn't the `back` link hidden at this point? Needed to change the text to Close?

Update: Nevermind, I see you are showing it in `anyCallback();`

**Comment by MSusik** [comment]
@jmartinm Implemented your suggestions.

**Comment by kaplun** [comment]
OK I am going to deploy it then

**Pull request merged**